### PR TITLE
Update era

### LINKED ISSUE #48: US: User can update an existing era

### PR DESCRIPTION
**Pull request by leahriffell**
### Code Highlights:
 * Adds mutation and spec for updating an existing era
 * Nests existing update_user spec under users folder

### Where should the reviewer start?
 * update_era_spec.rb

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.

### Issues:
* Closes #48 

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations:

### CODE REVIEW

File: spec/requests/graphql/mutations/eras/update_era_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      era_id = user.eras[0].id
+
+      post graphql_path, params: { query: query(era_id) }
+      result = JSON.parse(response.body)
+
+      data = result['data']['updateEra']
+
+      expect(data['name']).to eq('Master chambray tote bag fixie')
+      expect(data['userId']).to eq(user.id.to_s)
+      expect(data['startDate']).to eq('2021-01-04 00:00:00 UTC')
```
Comment (leahriffell) [inline spec/requests/graphql/mutations/eras/update_era_spec.rb]: Once Luke's PR on changing dates from DateTime to Date is merged, I can change these assertions to remove times.

Comment (LHJE) [reply spec/requests/graphql/mutations/eras/update_era_spec.rb]: Looks Good!!

**Review by LHJE — APPROVED** [review]
Pretty standard changes at this point, amiright?  Looking good.

File: app/graphql/mutations/eras/update_era.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  argument :color, String, required: false
+
+  type Types::EraType
+
+  def resolve(attributes)
+    era = Era.find(attributes[:id])
+    era.update(attributes)
+    era
+  end
+end
```
Comment (AngelaGuardia) [inline app/graphql/mutations/eras/update_era.rb:15]: Nice!

**Review by AngelaGuardia — APPROVED** [review]

File: spec/requests/graphql/mutations/eras/update_era_spec.rb (hunk 1 above)
Comment (dhalverson) [reply spec/requests/graphql/mutations/eras/update_era_spec.rb]: Luke's PR has been merged!

**Review by dhalverson — APPROVED** [review]
Reviewed and approved. Leah should be good to go with the assertions

**Pull request merged**